app/migrations/Version20230621074925.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\Exception\SkipMigration;
+use Mautic\CoreBundle\Doctrine\AbstractMauticMigration;
+use Mautic\PointBundle\Entity\Group;
+use Mautic\PointBundle\Entity\GroupContactScore;
+
+final class Version20230621074925 extends AbstractMauticMigration
+{
+    /**
+     * @throws SkipMigration
+     */
+    public function preUp(Schema $schema): void
+    {
+        $shouldRunMigration = !$schema->hasTable($this->generateTableName(Group::TABLE_NAME));
+
+        if (!$shouldRunMigration) {
+            throw new SkipMigration('Schema includes this migration');
+        }
```

Comment:
We have bad experience with these massive migrations and simple preUp checks. It fails in the middle for 1 instance. So our dev ops team run it again. It succeeds because the preUp check says it's migrated. But it's missing some alter tables that has failed before. So this migration silently fails and then our customers have to report broken instances.

That's why we created https://github.com/mautic/mautic/blob/9a609d5b8d83ddfa27bb752a91b863738f60b12c/app/migrations/Migration.template#L12 so it at least fails and we know about it.

Reply:
Good to know 👍 I will update this migration